### PR TITLE
perf: parallelise all async calls

### PR DESCRIPTION
**Pull request by ineshbose**
- about ~15-20% faster build (tested in my use cases, can go up for 30-50% as noted in CI) 🔥 
- functional programming to ensure no side-effects ⚙️ 
- splits & organises files for easier development in future 📦 
- provides hook for each loaded config 👁️ 
- drops `requireModule` for `await import()` (also parallelised) 🚢

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -1,166 +1,80 @@
...
-        // only if the path is found
-        if (resolvedPath) {
-          configPaths.push(resolvedPath)
+    const [configPaths, contentPaths] = await resolveModulePaths(moduleOptions.configPath, nuxt)
+
+    const tailwindConfig = await Promise.all((
+      configPaths.map(async (configPath, idx, paths) => {
+        let _tailwindConfig: Partial<TWConfig> | undefined
+        try {
+          _tailwindConfig = await import(configPath).then(c => c.default || c)
```
Comment (ineshbose) [inline src/module.ts:55]: Can append `.catch` block but not required

File: src/viewer.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { underline, yellow } from 'colorette'
+import { eventHandler, sendRedirect } from 'h3'
+import { addDevServerHandler, isNuxt2, isNuxt3, useLogger, useNuxt } from '@nuxt/kit'
+import { withTrailingSlash, withoutTrailingSlash, joinURL } from 'ufo'
+import type { TWConfig } from './types'
+
+export default async function (twConfig: Partial<TWConfig>, nuxt = useNuxt()) {
+  // const { withTrailingSlash, withoutTrailingSlash, joinURL } = await import('ufo')
```
Comment (ineshbose) [inline src/viewer.ts:8]: My understanding from this is that the dynamic import is for treeshaking/bundle-size?

Although `ufo` is in the module dependencies as well as lots of Nuxt packages - I feel that it is negligible...

Comment (Atinux) [reply src/viewer.ts:8]: Agree we can avoid having `await import('ufo')` since it is anyway on the server part

Comment (Atinux) [inline src/viewer.ts:8]:
```suggestion
```

File: src/module.ts (hunk 1 above)
Comment (ineshbose) [reply src/module.ts:55]: Causing issues in build when TS config is imported into `module.mjs`

**Pull request merged**